apollo-api/src/main/java/com/apollostack/api/graphql/ResponseReader.java

```diff
@@ -0,0 +1,42 @@
+package com.apollostack.api.graphql;
+
+import java.util.List;
+
+/** TODO java doc **/
+public interface ResponseReader {
```

Comment:
Also it would be nice if you could break this PR down into multiple smaller ones if possible